Add decision about headers to dev docs (#3402)
SpikeInterface/spikeinterface: doc/development/development.rst

> #. Avoid using abreviations in variable names (e.g., use :code:`recording` instead of :code:`rec`). It is especially important to avoid single letter variables.

Author: ```suggestion
#. Avoid using abbreviations in variable names (e.g. use :code:`recording` instead of :code:`rec`). It is especially important to avoid single letter variables.
```
@zm711

#. Use index as singular and indices for plural following the NumPy convention. Avoid idx or indexes. Plus, id and ids are reserved for identifiers (i.e. channel_ids)
#. We use file_path and folder_path (instead of file_name and folder_name) for clarity.

> #. For creating headers to divide section of codes we use the following convention (see issue 3019):

Reviewer: ```suggestion
#. For creating headers to divide section of codes we use the following convention (see issue `#3019 <https://github.com/SpikeInterface/spikeinterface/issues/3019>`_):
```

> #. For creating headers to divide section of codes we use the following convention (see issue `#3019 <https://github.com/SpikeInterface/spikeinterface/issues/3019>`_):

Reviewer 2: ```suggestion
#. For creating headers to divide sections of code we use the following convention (see issue `#3019 <https://github.com/SpikeInterface/spikeinterface/issues/3019>`_):
```
